pkg/server/node.go

```diff
@@ -348,7 +348,7 @@ func (n *Node) start(
 	locality roachpb.Locality,
 	localityAddress []roachpb.LocalityAddress,
 	nodeDescriptorCallback func(descriptor roachpb.NodeDescriptor),
-) error {
+) (error, func()) {
```

Comment:
Let's use named return values here: https://tour.golang.org/basics/7. Let's also add some commentary above explaining what this new return value is intended for.